spack/wdmapp/packages/wdmapp/package.py

```diff
@@ -56,3 +52,21 @@ class Wdmapp(BundlePackage):
     # variant +tau
     depends_on('tau@develop +adios2 ~libunwind ~pdt +mpi', when='+tau')
 
+    # variant +effis
+    depends_on('effis python-type=minimal', when='+effis')
+    depends_on('gene@coupling +effis', when='~passthrough +effis')
+    depends_on('xgc-devel@wdmapp +effis', when='~passthrough +effis')
+
+
+    # FIXME these are hacks to avoid Spack not finding a feasible packages on its own
+    # CWS - If the variant is not specified for hdf5 then there are
+    # concretization errors associated with hdf5. I think this is related 
+    # to https://github.com/spack/spack/issues/15478 .
+
+    # Contretization fixes
+    depends_on('adios  +fortran +sz')
+    depends_on('adios2 +fortran +sz')   # The ADIOS +sz (+fortran) is to be explicit, but it's the default
+    depends_on('sz@:1.4.12.99')         # ADIOS1 needs sz < 2.0.0, ADIOS2 defaults to sz > 2.0.0
+    depends_on('hdf5 +hl +fortran')     # PETSc needes +hl, Apps need +fortran
+    depends_on('python@:2.9.99')        # PETSc 3.7.7 needs Python 2.7, EFFIS defaults to Python 3
```

Comment:
Did you try whether this works with spack v14.2? @cwsmith and I had troubles where we had to list these identical dependencies for each variant -- but maybe by listing them last, not first, spack works as intended? If so, that'd be good.

Reply:
I didn't try v0.14.2. There's definitely sometimes still oddities in what happens if specs aren't explicitly given everywhere. I wouldn't be surprised if you need to repeat for each variant. If you notice, I did that in the EFFIS package. Though, I think that was needed becasue `python-type=full` is being set from `packages.yaml`, and I don't think that fully settles out concretization issues in all cases.